### PR TITLE
[FIX] Support RFC-3848 service types in Received headers

### PR DESCRIPTION
**Pull request by chibenwa**
- Detail LMTP service type
 - Expose auth settings through service type
 - Extract message generation
 
 CF https://datatracker.ietf.org/doc/html/rfc3848

### CODE REVIEW

**Review by quantranhong1999 — COMMENTED** [review]
Read it

**Comment by chibenwa** [comment]
Ideally we shall write some tests for this new message generator

File: protocols/lmtp/src/main/java/org/apache/james/protocols/lmtp/core/ReceivedDataLineFilter.java [hunk 1]
```diff
@@ -19,19 +19,45 @@
...
+                        return SERVICE_TYPE;
+                    }
+                } else {
+                    // See RFC3848:
+                    // The new keyword "ESMTPA" indicates the use of ESMTP when the SMTP
+                    // AUTH [3] extension is also used and authentication is successfully achieved.
+                    if (session.isTLSStarted()) {
+                        return SERVICE_TYPE_SSL_AUTH;
+                    } else {
+                        return SERVICE_TYPE_SSL;
```
Comment (Arsnael) [inline protocols/lmtp/src/main/java/org/apache/james/protocols/lmtp/core/ReceivedDataLineFilter.java:56]: shouldn't we return here SERVICE_TYPE_AUTH ?

File: protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/ReceivedHeaderGenerator.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+public class ReceivedHeaderGenerator {
+    private static final DateTimeFormatter DATEFORMAT = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss Z (zzz)", Locale.US);
+
+    private static final String EHLO = "EHLO";
+    private static final String SMTP = "SMTP";
+    private static final String ESMTPA = "ESMTPA";
+    private static final String ESMTPSA = "ESMTPA";
+    private static final String ESMTP = "ESMTP";
+    private static final String ESMTPS = "ESMTP";
```
Comment (Arsnael) [inline protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/ReceivedHeaderGenerator.java:42]: Any reason that `ESMTPSA` constant is "ESMTPA" and not "ESMTPSA"?

Same question for `ESMTPS`

**Comment by chibenwa** [comment]
Typos sorry

File: protocols/lmtp/src/main/java/org/apache/james/protocols/lmtp/core/ReceivedDataLineFilter.java (hunk 1 above)
Comment (Arsnael) [reply protocols/lmtp/src/main/java/org/apache/james/protocols/lmtp/core/ReceivedDataLineFilter.java:56]: Was I wrong here?

**Comment by Arsnael** [comment]
I believe the last fixup was forgotten and necessary too. Please dont hesitate to tell me if it was wrong.

**Review by Arsnael — APPROVED** [review]

**Pull request merged**